src/redux/slices/uiSlice.ts

```diff
@@ -0,0 +1,71 @@
+import { PayloadAction, createSlice } from '@reduxjs/toolkit';
```

Comment:
why are we using `redux`

Reply:
redux toolkit is already implemented in UIKit. 
And I prefer redux rather than context for performance.